cql3/restrictions/statement_restrictions.cc

```diff
@@ -1364,6 +1364,43 @@ void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool
         const auto pos = _schema->position(*col) + 1;
         (*_idx_tbl_ck_prefix)[pos] = replace_column_def(e, &idx_tbl_schema.clustering_column_at(pos));
     }
+
+    add_clustering_restrictions_to_idx_ck_prefix(idx_tbl_schema);
+
+    auto token_column = &idx_tbl_schema.clustering_column_at(0);
+    (*_idx_tbl_ck_prefix)[0] = binary_operator(
+            column_value(token_column),
+            oper_t::EQ,
+            // TODO: This should be a unique marker whose value we set at execution time.  There is currently no
+            // handy mechanism for doing that in query_options.
+            ::make_shared<constants::value>(raw_value::make_unset_value()));
+}
+
+void statement_restrictions::prepare_indexed_local(const schema& idx_tbl_schema) {
+    if (!_partition_range_is_simple) {
+        return;
+    }
+
+    // Local index clustering key is (indexed column, base clustering key)
+    _idx_tbl_ck_prefix = std::vector<expr::expression>();
+    _idx_tbl_ck_prefix->reserve(1 + _clustering_prefix_restrictions.size());
+
+    const column_definition& indexed_column = idx_tbl_schema.column_at(column_kind::clustering_key, 0);
+    const column_definition& indexed_column_base_schema = *_schema->get_column_definition(indexed_column.name());
+
+    // Find index column restrictions in the WHERE clause
+    std::vector<expr::expression> idx_col_restrictions = extract_column_restrictions(*_where, indexed_column_base_schema);
+    expr::expression idx_col_restriction_expr = expr::expression(expr::conjunction{std::move(idx_col_restrictions)});
```

Comment:
Aha, so you're converting it to a conjunction. Why not return a conjuction directly?

Reply:
I thought returning a `vector` could be more general, maybe someone would like to take extract restrictions and filter out multi column ones or something

Reply:
Ok. They could explode the conjunction if they wanted to, though.